boa/src/builtins/string/mod.rs

```diff
@@ -647,10 +630,9 @@ impl String {
     /// [spec]: https://tc39.es/ecma262/#sec-string.prototype.match
     /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match
     /// [regex]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions
-    pub(crate) fn r#match(this: &mut Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
-        let mut re =
-            RegExp::make_regexp(&mut Value::from(Object::default()), &[args[0].clone()], ctx)?;
-        RegExp::r#match(&mut re, ctx.to_string(this)?, ctx)
+    pub(crate) fn r#match(this: &Value, args: &[Value], ctx: &mut Interpreter) -> ResultValue {
+        let re = RegExp::make_regexp(&Value::from(Object::default()), &[args[0].clone()], ctx)?;
```

Comment:
Am I wrong to think that this will panic if no arguments are passed to this function?

Reply:
Yes. it definitely will, out of bounds panic. I did not check that code, there are many places were it should not panic or just throw an exception, I did not focus on these parts, I wanted to do it in a separate PR.